server/dex/feemgr.go

```diff
@@ -104,3 +104,14 @@ func (f *feeFetcher) LastRate() uint64 {
 func (f *feeFetcher) MaxFeeRate() uint64 {
 	return f.Asset.MaxFeeRate
 }
+
+// SwapFeeRate returns the tx fee that needs to be used to initiate a swap.
+// This fee will be the max fee rate if the asset supports dynamic tx fees,
+// and otherwise it will be the current market fee rate.
+func (f *feeFetcher) SwapFeeRate(ctx context.Context) uint64 {
+	if f.Backend.SupportsDynamicTxFee() {
+		return f.MaxFeeRate()
+	} else {
+		return f.FeeRate(ctx)
```

Comment:
One thing to consider is for ETH does `eth.node.suggestGasPrice(ctx)` (part of `FeeRate` about 10 lines) even have much value anymore?  That function [gives a fee rate suitable for legacy (non-dynamic) txns](https://github.com/ethereum/go-ethereum/blob/015fde9a2c4b6e722a9de5bfea2a14226806a231/internal/ethapi/api.go#L63-L64), I had actually sorta figured we'd kill that and have `FeeRate` behave more like maxFeeRate.  I can see that being valuable as an oracle for clients, maybe, but we'd probably adjust the formula because I find it doesn't give high enough fee rates.

I think your approach works though.  Just scanned through once.

Reply:
Yeah I wanted to keep it so the `FeeRateRoute` wouldn't just return the max fee rate. How about 2 * previous base rate?

Reply:
:+1: That seems to be geth's default in various API functions.

https://github.com/ethereum/go-ethereum/blob/015fde9a2c4b6e722a9de5bfea2a14226806a231/internal/ethapi/transaction_args.go#L95-L98